halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/model/v1/canary/google/GoogleCanaryAccount.java

```diff
@@ -47,11 +46,8 @@
   private String rootFolder = "kayenta";
   private SortedSet<AbstractCanaryServiceIntegration.SupportedTypes> supportedTypes = new TreeSet<>();
 
-  @Autowired
-  private SecretSessionManager secretSessionManager;
-
   @JsonIgnore
-  public GoogleNamedAccountCredentials getNamedAccountCredentials(String version, ConfigProblemSetBuilder p) {
+  public GoogleNamedAccountCredentials getNamedAccountCredentials(String version, SecretSessionManager secretSessionManager, ConfigProblemSetBuilder p) {
```

Comment:
Should we null guard `secretSessionManager` as in `GoogleAccount`?

Reply:
The only place this method is getting called is in `GoogleCanaryAccountValidator` which has a (correctly) autowired `SecretSessionManager`. But if we want to be safe, I can certainly add that in.

Reply:
SG as is. Probably better to fail explicitly than default to subtle backup behavior.